Default to Kubernetes 1.24

skitt opened the pull request:
Signed-off-by: Stephen Kitt <skitt@redhat.com>

Depends on https://github.com/submariner-io/lighthouse/pull/791
Depends on https://github.com/submariner-io/admiral/issues/388

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

skitt committed:
Default to Kubernetes 1.24

Signed-off-by: Stephen Kitt <skitt@redhat.com>

skitt commented:
The upgrade job won’t work until we release 0.13.0 or 0.12.2. (0.12.1 doesn’t support Kubernetes 1.24.)

dfarrell07 commented:
The Helm jobs will fail because of https://github.com/submariner-io/submariner-charts/issues/236.

mkolesnik commented on scripts/shared/clusters.sh:
```diff
@@ -29,6 +46,7 @@ eval set -- "${FLAGS_ARGV}"
 k8s_version="${FLAGS_k8s_version}"
 olm_version="${FLAGS_olm_version}"
 [[ -z "${k8s_version}" ]] && k8s_version="${DEFAULT_K8S_VERSION}"
+kind="${kind_kind_binaries[$k8s_version]}"
```
```suggestion
kind="${kind_kind_binaries[$k8s_version]:-kind}"
```

Wouldn't this work, thus obviating the need to declare kind binaries for 1.17-1.23?

Also it would fit well with 
> kind-0.12 is the default

skitt replied:
Yes, the last change was a quick fix to see if I could get the whole set of jobs passing.

mkolesnik commented on scripts/shared/clusters.sh:
```diff
@@ -2,15 +2,32 @@
-DEFAULT_K8S_VERSION=1.23
-declare -A kind_k8s_versions
+DEFAULT_K8S_VERSION=1.24
+declare -A kind_k8s_versions kind_kind_binaries
```
```suggestion
declare -A kind_k8s_versions kind_binary
```

Clearer like this and less repetitive

mkolesnik commented on scripts/shared/clusters.sh:
```diff
@@ -37,7 +55,7 @@ olm_version="${FLAGS_olm_version}"
 timeout="${FLAGS_timeout}"
 
-echo "Running with: k8s_version=${k8s_version}, olm_version=${olm_version}, olm=${olm}, globalnet=${globalnet}, prometheus=${prometheus}, registry_inmemory=${registry_inmemory}, settings=${settings}, timeout=${timeout}"
+echo "Running with: kind=${kind}, k8s_version=${k8s_version}, olm_version=${olm_version}, olm=${olm}, globalnet=${globalnet}, prometheus=${prometheus}, registry_inmemory=${registry_inmemory}, settings=${settings}, timeout=${timeout}"
```
I wouldn't add this here, this printout is for user supplied variables.

The kind binary used will be visible on the debug print anyhow, especially:
```
[14:56:55.391] [shipyard]$ [cluster1] kind-0.14 version
[cluster1] kind v0.14.0 go1.18.2 linux/amd64
```

mkolesnik commented on scripts/shared/clusters.sh:
```diff
@@ -118,9 +136,9 @@ function create_kind_cluster() {
     rm -f "$KUBECONFIG"
 
-    if kind get clusters | grep -q "^${cluster}$"; then
+    if ${kind} get clusters | grep -q "^${cluster}$"; then
```
```suggestion
    if "${kind}" get clusters | grep -q "^${cluster}$"; then
```
For shellcheck? (same for other calls)

mkolesnik commented on scripts/shared/clusters.sh:
```diff
@@ -2,15 +2,32 @@
 kind_k8s_versions[1.22]=1.22.7@sha256:1dfd72d193bf7da64765fd2f2898f78663b9ba366c2aa74be1fd7498a1873166
 kind_k8s_versions[1.23]=1.23.4@sha256:0e34f0d0fd448aa2f2819cfd74e99fe5793a6e4938b328f657c8e3f81ee0dfb9
+# kind-0.14 hashes
+#kind_k8s_versions[1.18]=1.18.20@sha256:738cdc23ed4be6cc0b7ea277a2ebcc454c8373d7d8fb991a7fcdbd126188e6d7
+#kind_k8s_versions[1.19]=1.19.16@sha256:d9c819e8668de8d5030708e484a9fdff44d95ec4675d136ef0a0a584e587f65c
+#kind_k8s_versions[1.20]=1.20.15@sha256:6f2d011dffe182bad80b85f6c00e8ca9d86b5b8922cdf433d53575c4c5212248
+#kind_k8s_versions[1.21]=1.21.12@sha256:f316b33dd88f8196379f38feb80545ef3ed44d9197dca1bfd48bcb1583210207
+#kind_k8s_versions[1.22]=1.22.9@sha256:8135260b959dfe320206eb36b3aeda9cffcb262f4b44cda6b33f7bb73f453105
+#kind_k8s_versions[1.23]=1.23.6@sha256:b1fa224cc6c7ff32455e0b1fd9cbfd3d3bc87ecaa8fcb06961ed1afb3db0f9ae
```
```suggestion
```

Wouldn't add these as they'll change anyway when KIND release updated images

skitt commented:
@mkolesnik I’ve addressed your review comments in https://github.com/submariner-io/shipyard/pull/864 (the code you were reviewing was already merged).

skitt commented:
The Helm jobs are now fixed.

@mkolesnik this is blocked by your request for changes, which doesn’t apply to this PR.

tpantelis commented:
I think we want to hold off on this until the lighthouse E2E resource issue 0.24 is resolved.

skitt commented:
> I think we want to hold off on this until the lighthouse E2E resource issue 0.24 is resolved.

Could you add that as a “request for changes” so it blocks the merge? Thanks! (I can’t do it since this is my own PR.)

tpantelis reviewed (changes requested):
Let's hold this until the lighthouse issues are fixed.

mkolesnik commented:
> Could you add that as a “request for changes” so it blocks the merge? Thanks! (I can’t do it since this is my own PR.)

I added it as "depends on", isn't that the preferred way of doing it? :)

FWIW on LH this change won't affect the CI, but obviously when deployed manually one could potentially hit some issues, so I guess it makes sense to wait on LH CI to pass to be sure.

We can also see jobs failing "E2E Consuming", including admiral which I think merits an investigation.

skitt commented:
> > Could you add that as a “request for changes” so it blocks the merge? Thanks! (I can’t do it since this is my own PR.)
> 
> I added it as "depends on", isn't that the preferred way of doing it? :)
> 
> FWIW on LH this change won't affect the CI, but obviously when deployed manually one could potentially hit some issues, so I guess it makes sense to wait on LH CI to pass to be sure.
> 
> We can also see jobs failing "E2E Consuming", including admiral which I think merits an investigation.

I was thinking of the latter, and since they aren’t “Required” (the Lighthouse ones are failing, as well as Admiral), I wanted to avoid an accidental merge before they were fixed.

dfarrell07 reviewed (changes requested):
Requesting changes until the not-required `Consuming Projects / E2E Consuming (admiral, operator, libreswan, 1.24)` job is passing.

tpantelis committed:
Merge branch 'devel' into default-k8s-1.24

tpantelis commented:
> Requesting changes until the not-required `Consuming Projects / E2E Consuming (admiral, operator, libreswan, 1.24)` job is passing.

It's passing now.

skitt commented:
@dfarrell07 could you update your review?

dfarrell07 committed:
Merge branch 'devel' into default-k8s-1.24
